File: Lista/Java/Q9.java
interface Imprimivel {
    void imprimir();
}

class Relatorio implements Imprimivel {
    private String titulo;
    private String conteudo;

    public Relatorio(String titulo, String conteudo) {
        this.titulo = titulo;
        this.conteudo = conteudo;
    }

    @Override
    public void imprimir() {
        System.out.println("----- Relatório -----");
        System.out.println("Título: " + titulo);
        System.out.println("Conteúdo: " + conteudo);
        System.out.println("---------------------");
    }
}

class Contrato implements Imprimivel {
    private String parteA;
    private String parteB;

    public Contrato(String parteA, String parteB) {
        this.parteA = parteA;
        this.parteB = parteB;
    }

    @Override
    public void imprimir() {
        System.out.println("----- Contrato -----");
        System.out.println("Parte A: " + parteA);
        System.out.println("Parte B: " + parteB);
        System.out.println("---------------------");
    }
}

public class Main {
    public static void main(String[] args) {
        Imprimivel relatorio = new Relatorio("Relatório Mensal", "Este é o conteúdo do relatório mensal.");
        Imprimivel contrato = new Contrato("Empresa XYZ", "Serviço de Consultoria");

        relatorio.imprimir();
        System.out.println();
        contrato.imprimir();
    }
}

File: Lista/Python/Q9.py
from abc import ABC, abstractmethod

class Imprimivel(ABC):
    
    @abstractmethod
    def imprimir(self):
        pass

class Relatorio(Imprimivel):
    def __init__(self, titulo, conteudo):
        self.titulo = titulo
        self.conteudo = conteudo

    def imprimir(self):
        return f'Relatório: {self.titulo}\nConteúdo: {self.conteudo}'

class Contrato(Imprimivel):
    def __init__(self, partes, termos):
        self.partes = partes
        self.termos = termos

    def imprimir(self):
        return f'Contrato entre: {self.partes}\nTermos: {self.termos}'

relatorio = Relatorio('Vendas Mensais', 'Relatório das vendas do mês de setembro')
contrato = Contrato('Empresa X e Empresa Y', 'Termos de prestação de serviços')

print(relatorio.imprimir())
print(contrato.imprimir())
